website/static/js/share/results.js

```diff
@@ -151,8 +151,18 @@ var Contributor = {
             m('a', {
                 href: '#',
                 onclick: function() {
-                    utils.updateFilter(vm, 'match:contributors.familyName:' + contributor.familyName, true);
-                    utils.updateFilter(vm, 'match:contributors.givenName:' + contributor.givenName, true);
+                    var givenNameLength = contributor.givenName ? contributor.givenName.length : 0;
+                    var familyNameLength = contributor.familyName ? contributor.familyName.length : 0;
+                    if (givenNameLength <= 0 && familyNameLength <= 0) {
```

Comment:
Hopefully we'll never see a length < 0; perhaps use `=== 0` instead. 

You could alternately shorten this to  `if ( !givenNameLength && !familyNameLength)...` ...Though I admit to finding the numeric comparison a bit more readable.